### PR TITLE
fix(lineage): Fix issue where downstream of datajobs do not appear

### PR DESCRIPTION
**Pull request by dexter-mh-lee**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-io/src/main/java/com/linkedin/metadata/graph/LineageRegistry.java [hunk 1]
```diff
@@ -52,14 +52,14 @@ public LineageRegistry(EntityRegistry entityRegistry) {
     // A produces B :     A -> downstream (produces, OUTGOING), B -> upstream (produces, INCOMING)
     for (LineageEdge edge : lineageEdges) {
       if (edge.isUpstream()) {
-        upstreamPerEntity.computeIfAbsent(edge.sourceEntity, (k) -> new HashSet<>())
+        upstreamPerEntity.computeIfAbsent(edge.sourceEntity.toLowerCase(), (k) -> new HashSet<>())
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/graph/LineageRegistry.java:55]: Why is this required?

Comment (dexter-mh-lee) [reply metadata-io/src/main/java/com/linkedin/metadata/graph/LineageRegistry.java:55]: It's because sometimes we use lowercased entity name and sometimes we don't so need to make sure we lower case on read and write to this map, so both are handled correctly. I missed putting toLowerCase on the write which caused issues only on dataJobs (vs datajobs) Added test cases to cover dataJobs

**Review by shirshanka — APPROVED** [review]
LGTM
